### PR TITLE
Feature/sorted

### PR DESCRIPTION
**Pull request by jlconlin**
This branch is the beginning of a namespace used to verify various pieces of the data. In particular, we've created some classes to verify that a range of data is sorted.

### CODE REVIEW

**Comment by apmccartney** [comment]
This build breaks coverage. Please revert your changes to the travis files. See the current master for working files

File: src/ACEtk/details/verify/Sorted/src/check.hpp [hunk 1]
```diff
@@ -0,0 +1,15 @@
+template< typename Arg,
+          utility::Require< true, utility::is_range, Range> = true >
+decltype( auto ) check( Arg&& arg ){
+  auto sortedEnd = ranges::is_sorted_until( arg );
+  if( sortedEnd != arg.end() ){
+    auto index = sortedEnd - arg.begin();
```
Comment (apmccartney) [inline src/ACEtk/details/verify/Sorted/src/check.hpp]: prefer `std::distance` or `ranges::distance`

File: src/ACEtk/details/verify/Sorted/src/ctor.hpp [hunk 2]
```diff
@@ -0,0 +1,24 @@
...
+        >
+Sorted( Arg&& arg ) :
+  Range( check( std::forward< Arg >( arg ) ) )
+{ }
+
+template< typename... Args, 
+          utility::Require< true, std::is_constructible, Range, Args... > = true
+        >
+Sorted( Args&&... args ):
+  Range( check( std::forward< Args >( args )... ) )
```
Comment (apmccartney) [inline src/ACEtk/details/verify/Sorted/src/ctor.hpp]: This doesn't make sense. Check cannot be applied to multiple arguments. You can only verify sorted-ness after constructing the parent. See the lazy verification in the example I sent.

File: src/ACEtk/details/verify/test/verify.test.cpp [hunk 3]
```diff
@@ -0,0 +1,9 @@
+#define CATCH_CONFIG_MAIN
+#include <chrono>
```
Comment (apmccartney) [inline src/ACEtk/details/verify/test/verify.test.cpp]: The only header you need here is catch

File: src/ACEtk/details/verify/src/sorted.hpp [hunk 4]
```diff
@@ -0,0 +1,15 @@
+template< typename Range,
+          utility::Require< false, IsSorted, Range > = true,
+          utility::Require< true, utility::is_range, Range > = true 
+        >
+auto sorted( Range&& r ){
+  return Sorted< std::decay_t< Range > >{ std::forward< Range >( r ) };
+}
+
+template< typename Range,
+          utility::Require< false, IsSorted, Range > = true
```
Comment (apmccartney) [inline src/ACEtk/details/verify/src/sorted.hpp]: the is only safe if the range is sorted. Here you're enforcing that it's not sorted.

**Review by apmccartney — CHANGES_REQUESTED** [review]
See inline comments. The lack of coverage information is a show stopper.

File: src/ACEtk/details/verify/test/verify.test.cpp (hunk 3 above)
Comment (jlconlin) [reply src/ACEtk/details/verify/test/verify.test.cpp]: That must have been copied/pasted from somewhere.

File: src/ACEtk/details/verify/src/sorted.hpp (hunk 4 above)
Comment (jlconlin) [reply src/ACEtk/details/verify/src/sorted.hpp]: Good catch. Another copy/paste error.

**Comment by jlconlin** [comment]
The travis configuration seems to work well. See [here](https://travis-ci.org/njoy/ACEtk/builds/406429555). I did copy from master, with a few minor modifications.

**Comment by apmccartney** [comment]
The travis configuration changes dropped the coverage value, which was working on recent previous pull requests

File: src/ACEtk/details/verify/Sorted/src/ctor.hpp (hunk 2 above)
Comment (apmccartney) [reply src/ACEtk/details/verify/Sorted/src/ctor.hpp]: Aside from the coverage, I think this is the last thing to be addressed.

**Comment by jlconlin** [comment]
This PR is ready to be looked at (one last time, I hope)

**Review by apmccartney — APPROVED** [review]
Excellent work Jeremy

**Pull request merged**